src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,123 @@
+import './Carousel.module.scss';
+
+import React from 'react';
+import { useEffect, useState } from 'react';
+
+import LeftArrMobile from '../../assets/svgs/LeftArrMobile.svg';
+import RightArrMobile from '../../assets/svgs/RightArrMobile.svg';
+import {
+  arrowContainer,
+  carousel,
+  img,
+  indicator,
+  indicatorFill,
+  indicators,
+  inner,
+  leftArrow,
+  rightArrow,
+  slider,
+} from './Carousel.module.scss';
+import ph from './imgs/ph.jpg';
+import ph1 from './imgs/ph1.jpg';
+import ph2 from './imgs/ph2.jpg';
+import ph3 from './imgs/ph3.jpg';
+
+export const Carousel = () => {
+  const images = [ph, ph1, ph2, ph3];
+  const length = images.length;
+
+  const [index, setIndex] = useState(0);
+  const [isClickable, setClickable] = useState(true);
+
+  const handlePrevious = () => {
+    if (isClickable) {
+      const newIndex = index - 1;
+      setIndex(newIndex < 0 ? length - 1 : newIndex);
+
+      setClickable(false);
+
+      setTimeout(() => {
+        setClickable(true);
+      }, 500);
+    }
+  };
+
+  const handleNext = () => {
+    if (isClickable) {
+      const newIndex = index + 1;
+      setIndex(newIndex >= length ? 0 : newIndex);
+
+      setClickable(false);
+
+      setTimeout(() => {
+        setClickable(true);
+      }, 500);
+    }
+  };
+
+  const setImage = (index) => {
+    if (isClickable) {
+      setIndex(index);
+
+      setClickable(false);
+
+      setTimeout(() => {
+        setClickable(true);
+      }, 500);
+    }
+  };
+
+  useEffect(() => {
+    const sliderTimeout = setTimeout(() => {
+      const newIndex = index + 1;
+      setIndex(newIndex >= length ? 0 : newIndex);
```

Comment:
Gdy zmieniamy stan na bazie stanu poprzedniego dobrym podejściem jest użycie notacji
 `setIndex(prev => prev + 1)` 
 1. Niweluje to ryzyko błędów i niepoprawnie zmienianych wartości
 2. Zwiększa wydajność aplikacji
 
 Więc w tym przykładzie użycie tego mogłoby wyglądać tak:
 ```
 setIndex(prev => prev + 1 >= length ? 0 : prev + 1);
 ```
 
 Zmień proszę resztę podobnych zmian stanów w kodzie na taką notację